ml-agents/mlagents/trainers/trainer_controller.py

```diff
@@ -263,6 +285,22 @@ def start_learning(self, env: BaseUnityEnvironment, trainer_config):
             self._write_training_metrics()
             self._export_graph()
 
+    def end_trainer_episodes(self, env, lessons_incremented):
+        curr_info = self._reset_env(env)
+        for brain_name, trainer in self.trainers.items():
+            trainer.end_episode()
+        for brain_name, changed in lessons_incremented.items():
+            if changed:
+                self.trainers[brain_name].reward_buffer.clear()
+        return curr_info
+
+    def check_empty_sampler_manager(self):
+        """
+        If self.samplers is empty, then bool of it returns false, indicating
+        there is no sampler manager.
+        """
+        return not bool(self.sampler_manager.samplers)
```

Comment:
How about making this a method or property on SamplerManager instead?

Reply:
+1 on making `check_empty_sampler_manager` part of SamplerManager

Reply:
Made the check part of the SamplerManager class